### DISCUSSION TITLE
Post-checkout lookalike for git repos?

### DISCUSSION THREAD

**Discussion by max06** [argoproj/argo-cd#17075] [Q&A]
I want to define an argo application based on an untracked path in a git repository. Said path gets generated running a command post checkout.

Example can be found [here](https://github.com/max06/deployments/tree/vendir-demo). `vendir.yaml` contains references to external sources that should be included at runtime, by running `vendir sync --locked`. In this case, the helm chart for pihole would be pulled and extracted to `vendor/pihole`, without committing it to the repo.

Can I tell argo to run said command before doing anything?

**Comment by jgwest** [comment]
Best I can suggest is using a custom [config management plugin](https://argo-cd.readthedocs.io/en/stable/operator-manual/config-management-plugins/), which would contain the commands you wish to run: On Argo sync, it would clone the repo, then call your script/command, and using stdout as the K8s resources.

**Reply by max06** [reply]
Thanks for that. That would certainly work, but I'd also have to use the git root directory in the app config since argo doesn't allow specifying a non-existent path. 

I'll see how far I get with that!
